File: java/src/main/java/vsdk/toolkit/common/statistics/Statistics.java
package vsdk.toolkit.common.statistics;

public class Statistics {
    public ReaderStatistics reader;
    public RadianceStatistics radiance;
    public PotentialStatistics potential;
    public ShadowStatistics shadow;
    public RayTracerStatistics rayTracer;

    private static Statistics instanceValue;

    public Statistics() {
        reader = new ReaderStatistics();
        radiance = new RadianceStatistics();
        potential = new PotentialStatistics();
        shadow = new ShadowStatistics();
        rayTracer = new RayTracerStatistics();
    }

    public static Statistics instance() {
        if (instanceValue == null) {
            instanceValue = new Statistics();
        }
        return instanceValue;
    }
}

File: typescript/base/src/main/vsdk/toolkit/common/statistics/Statistics.ts
import { PotentialStatistics } from "./PotentialStatistics";
import { RadianceStatistics } from "./RadianceStatistics";
import { RayTracerStatistics } from "./RayTracerStatistics";
import { ReaderStatistics } from "./ReaderStatistics";
import { ShadowStatistics } from "./ShadowStatistics";

export class Statistics {
  public reader: ReaderStatistics;
  public radiance: RadianceStatistics;
  public potential: PotentialStatistics;
  public shadow: ShadowStatistics;
  public rayTracer: RayTracerStatistics;

  private static instanceValue: Statistics | null = null;

  public constructor() {
    this.reader = new ReaderStatistics();
    this.radiance = new RadianceStatistics();
    this.potential = new PotentialStatistics();
    this.shadow = new ShadowStatistics();
    this.rayTracer = new RayTracerStatistics();
  }

  public static instance(): Statistics {
    if (Statistics.instanceValue === null) {
      Statistics.instanceValue = new Statistics();
    }
    return Statistics.instanceValue;
  }
}
